packages/webdriverio/src/scripts/isElementDisplayed.ts

```diff
@@ -114,25 +114,38 @@ export default function isElementDisplayed (element: Element): boolean {
         return cascadedStylePropertyForElement(parentElement, property)
     }
 
-    function elementHasBoundingBox(element: Element): boolean {
+    function elementSubtreeHasNonZeroDimension({
+        element,
+        dimension,
+    }: {
+        element: Element;
+        dimension: 'x' | 'y';
+    }): boolean {
         const boundingBox = element.getBoundingClientRect()
-        return boundingBox.width > 0 && boundingBox.height > 0
-    }
-
-    function elementSubtreeHasNonZeroDimensions(element: Element): boolean {
-        if (elementHasBoundingBox(element)) {
+        if (boundingBox.width > 0 && boundingBox.height > 0) {
             return true
         }
 
         // Paths can have a zero width or height. Treat them as shown if the stroke width is positive.
-        const boundingBox = element.getBoundingClientRect()
-        if (element.tagName.toUpperCase() === 'PATH' && boundingBox.width + boundingBox.height > 0) {
-            const strokeWidth = cascadedStylePropertyForElement(element, 'stroke-width')
-            return !!strokeWidth && (parseInt(strokeWidth, 10) > 0)
+        if (
+            element.tagName.toUpperCase() === 'PATH' &&
+            (boundingBox.width > 0 || boundingBox.height > 0)
```

Comment:
Shouldn't this be && instead of ||?

Reply:
This replaces `boundingBox.width + boundingBox.height > 0` for clarity, so `||` is appropriate